rust/core-lib/src/editor.rs

```diff
@@ -686,6 +686,44 @@ impl Editor {
             self.add_delta(builder.build());
         }
     }
+  
+    /// Changes the number(s) under the cursor(s) with the `transform_function`.
+    /// If there is a number next to or on the beginning of the region, then
+    /// this number will be replaced with the result of `transform_function` and
+    /// the cursor will be placed at the end of the number.
+    /// Some Examples with a increment `transform_function`:
+    ///
+    /// "|1234" -> "1235|"
+    /// "12|34" -> "1235|"
+    /// "-|12" -> "-11|"
+    /// "another number is 123|]" -> "another number is 124"
+    ///
+    /// This function also works fine with multiple regions.
+    fn change_number<F: Fn(i128) -> Option<i128>>(&mut self, view: &View,
+                                        transform_function: F) {
+        let mut builder = delta::Builder::new(self.text.len());
+        for region in view.sel_regions() {
+
+            let mut cursor = WordCursor::new(&self.text, region.end);
+            let (mut start, end) = cursor.select_word();
+
+            // see if the word begins with a -
+            if start > 0 && self.text.slice_to_cow(start - 1..start) == "-" {
```

Comment:
So I don't think this is a problem now, but at some point we will likely want to implement some approximation of [UAX29](http://unicode.org/reports/tr29/) which means it's likely there are characters that will be word boundaries that will have multi-byte representations in utf8; if that was the case this `slice` would panic, because `start - 1` would not be a word boundary. 

What we probably want to do is to explicitly grab the preceding codepoint boundary, and check that character. This is all handled by the generic `Cursor` type in the rope crate, but there's a convenience method on `Rope` itself; you can just do `rope.prev_codepoint_offset(start)`. **edit** I'll leave that there for informative purposes, but there's a better way to do this: you can just do `rope.byte_at(start - 1)`; this won't panic if we're not on a char boundary, and you can just compare the bytes directly.

Reply:
Thanks for the feedback! I'll give this a try later!

Reply:
Added it, works great.